assembly/src/parsers/blocks.rs

```diff
@@ -251,22 +254,45 @@ impl BlockParser {
 // ================================================================================================
 
 pub fn combine_blocks(mut blocks: Vec<CodeBlock>) -> CodeBlock {
-    // merge consecutive Span blocks
+    // merge consecutive Span blocks.
     let mut merged_blocks: Vec<CodeBlock> = Vec::with_capacity(blocks.len());
+    // Keep track of all the consecutive Span blocks and are merged together when there is a discontinuity.
```

Comment:
very minor nit: I try to keep the length of lines to 100 chars. It is fine to split the comment into two lines here.

Reply:
Done!